gno.land/pkg/sdk/vm/convert.go

```diff
@@ -153,6 +140,34 @@ func convertArgToGno(arg string, argT gno.Type) (tv gno.TypedValue) {
 			}
 			tv.SetUint64(u64)
 			return
+		case gno.Float32Type:
+			assertCharNotPlus(arg[0])
+			dec, _, err := apd.NewFromString(arg)
+			if err != nil {
+				panic(fmt.Sprintf("error parsing float32 %q: %v", arg, err))
+			}
+
+			f32, err := strconv.ParseFloat(dec.String(), 32)
```

Comment:
why the code to parse floats is different? We can avoid duplicating code doing the following:

``` go
func parseFloat(arg string, bitSize int) float64 {
	assertCharNotPlus(arg[0])
	dec, _, err := apd.NewFromString(arg)
	if err != nil {
		panic(fmt.Sprintf("error parsing float32 %q: %v", arg, err))
	}

	f, err := strconv.ParseFloat(dec.String(), bitsize)
	if err != nil {
		panic(fmt.Sprintf("error value exceeds float%d precision %q: %v",bitSize, arg, err))
	}

	return f
}
```

Reply:
The `apd.Decimal` type has a `Float64` method but not a `Float32` method.  I thought it might be better to use that method if it is available and fall back to `strconv.ParseFloat` for `float32` types.

I've just thought about this and can't think of any reason why not to do what you're suggesting here as it seems to present a cleaner solution, so I'll make that change.